k8s_test_harness/util/docker_util.py

```diff
@@ -4,11 +4,35 @@
 #
 
 import subprocess
+from typing import List
 
 
 def ensure_image_contains_paths(image, paths):
-    for path in paths:
-        subprocess.run(
-            ["docker", "run", "--rm", image, "ls", "-l", path],
-            check=True,
-        )
+    """Ensures the given container image contains the provided paths
+    within it by attempting to run the image and 'ls' all of them.
+    """
+    base_cmd = ["docker", "run", "--rm", image, "ls", "-l"]
+    base_cmd.extend(paths)
+
+    subprocess.run(base_cmd, check=True)
```

Comment:
nit: you can use https://github.com/canonical/k8s-test-harness/blob/d503366f0b481183567de806175ac8d6fee547dc/k8s_test_harness/util/exec_util.py#L25 for this.

Reply:
and below.

Reply:
Sadly there's something weird with how pytest loads modules going on and it [leads to partially initalized import issues](https://github.com/aznashwan/harbor-rocks/actions/runs/9989824882/job/27609690604#step:7:53).

In the interest of time, I have converted it back to using `subprocess` directly. (the added logging that `exec_util.run()` offers won't be printed on successful tests anyway, and the whole command line argument list will be reported by `subprocess` on errors anyway, so there's no real notable disadvantage without it anyway)